x509/errors.go

```diff
@@ -5,13 +5,279 @@ import "fmt"
 // To preserve error IDs, only append to this list, never insert.
 const (
 	ErrInvalidID ErrorID = iota
+	ErrInvalidCertList
+	ErrTrailingCertList
+	ErrUnexpectedlyCriticalCertListExtension
+	ErrUnexpectedlyNonCriticalCertListExtension
+	ErrInvalidCertListAuthKeyID
+	ErrTrailingCertListAuthKeyID
+	ErrInvalidCertListIssuerAltName
+	ErrInvalidCertListCRLNumber
+	ErrTrailingCertListCRLNumber
+	ErrNegativeCertListCRLNumber
+	ErrInvalidCertListDeltaCRL
+	ErrTrailingCertListDeltaCRL
+	ErrNegativeCertListDeltaCRL
+	ErrInvalidCertListIssuingDP
+	ErrTrailingCertListIssuingDP
+	ErrCertListIssuingDPMultipleTypes
+	ErrCertListIssuingDPInvalidFullName
+	ErrInvalidCertListFreshestCRL
+	ErrInvalidCertListAuthInfoAccess
+	ErrTrailingCertListAuthInfoAccess
+	ErrUnhandledCriticalCertListExtension
+	ErrUnexpectedlyCriticalRevokedCertExtension
+	ErrUnexpectedlyNonCriticalRevokedCertExtension
+	ErrInvalidRevocationReason
+	ErrTrailingRevocationReason
+	ErrInvalidRevocationInvalidityDate
+	ErrTrailingRevocationInvalidityDate
+	ErrInvalidRevocationIssuer
+	ErrUnhandledCriticalRevokedCertExtension
 
 	ErrMaxID
 )
 
 // idToError gives a template x509.Error for each defined ErrorID; where the Summary
 // field may hold format specifiers that take field parameters.
-var idToError = map[ErrorID]Error{}
+var idToError = map[ErrorID]Error{
+
+	ErrInvalidCertList: Error{
```

Comment:
Just a thought - if you populated this map in a `func init()` from a list of Error structs you could avoid the possibility of a mismatched id -> Error mapping (i.e. you're currently relying on the thing to the left of the `:` being the same as the thing to the right of `ID:` in the struct).

Reply:
Done.